pkg/relayer/proxy/proxy.go

```diff
@@ -59,60 +69,79 @@ func NewRelayerProxy(
 	clientCtx sdkclient.Context,
 	keyName string,
 	keyring keyring.Keyring,
-
+	nativeServicesListenAddress map[serviceId]string,
 	// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
 	// blockClient blocktypes.BlockClient,
 ) RelayerProxy {
 	accountQuerier := accounttypes.NewQueryClient(clientCtx)
 	supplierQuerier := suppliertypes.NewQueryClient(clientCtx)
 	sessionQuerier := sessiontypes.NewQueryClient(clientCtx)
-	providedServices := buildProvidedServices(ctx, supplierQuerier)
 	servedRelays, servedRelaysProducer := channel.NewObservable[*types.Relay]()
 
 	return &relayerProxy{
 		// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
-		// blockClient:          blockClient,
-		keyName:              keyName,
-		keyring:              keyring,
-		accountsQuerier:      accountQuerier,
-		supplierQuerier:      supplierQuerier,
-		sessionQuerier:       sessionQuerier,
-		providedServices:     providedServices,
-		servedRelays:         servedRelays,
-		servedRelaysProducer: servedRelaysProducer,
+		// blockClient:       blockClient,
+		keyName:                     keyName,
+		keyring:                     keyring,
+		accountsQuerier:             accountQuerier,
+		supplierQuerier:             supplierQuerier,
+		sessionQuerier:              sessionQuerier,
+		nativeServicesListenAddress: nativeServicesListenAddress,
+		servedRelays:                servedRelays,
+		servedRelaysProducer:        servedRelaysProducer,
 	}
 }
 
-// Start starts all supported proxies and returns an error if any of them fail to start.
+// Start concurrently starts all advertised relay servers and returns an error if any of them fails to start.
 func (rp *relayerProxy) Start(ctx context.Context) error {
```

Comment:
The other `Start` function has a comment that it's a blocking function that needs to be run in a goroutine. Does the same thing apply here?